### ISSUE TITLE
False positive in new_without_default: should not trigger when `new` is unsafe

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/rust-clippy#2875]
```rust
struct A;

pub struct B(A);

impl B {
    /// # Safety
    ///
    /// Must only be called after you have frobbed the other thing, or else nasal
    /// demons. Also you must ensure that this is called a prime number of times
    /// during each launch.
    pub unsafe fn new() -> Self {
        B(A)
    }
}
```

```
warning: you should consider adding a `Default` implementation for `B`
 --> src/lib.rs
  |
  | /     pub unsafe fn new() -> Self {
  | |         B(A)
  | |     }
  | |_____^
  |
  = note: #[warn(new_without_default)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.210/index.html#new_without_default
help: try this
  |
  | impl Default for B {
  |     fn default() -> Self {
  |         Self::new()
  |     }
  | }
  |
```

I would expect this not to trigger `new_without_default` because the signature of `fn default() -> Self` and `unsafe fn new() -> Self` are not the same.

**Comment by phansch** [comment]
Fixed via #3280
